src/NLog/LogReceiverService/WcfLogReceiverClient.cs

```diff
@@ -237,12 +265,22 @@ protected override ILogReceiverClient CreateChannel()
         private IAsyncResult OnBeginProcessLogMessages(object[] inValues, AsyncCallback callback, object asyncState)
         {
             var events = (NLogEvents)inValues[0];
-            return ((ILogReceiverClient)this).BeginProcessLogMessages(events, callback, asyncState);
+            if (UseOneWayCallsToServer) {
+                return ((ILogReceiverClient)this).BeginProcessLogMessagesV2(events, callback, asyncState);
+            }
+            else {
+                return ((ILogReceiverClient)this).BeginProcessLogMessages(events, callback, asyncState);
+            }
         }
 
         private object[] OnEndProcessLogMessages(IAsyncResult result)
         {
-            ((ILogReceiverClient)this).EndProcessLogMessages(result);
+            if (UseOneWayCallsToServer) {
```

Comment:
We don't need to copy the `if (UseOneWayCallsToServer)` (code duplication). I advise a helper method who's returning a `ILogReceiverClient`